ptf/saitest.py

```diff
@@ -0,0 +1,165 @@
+            vr_id=self.default_vrf)
+        status = sai_thrift_create_route_entry(
+            self.client, self.route3, next_hop_id=self.nhop)
+        self.assertEqual(status, SAI_STATUS_SUCCESS)
+        self.route4 = sai_thrift_route_entry_t(
+            switch_id=self.switch_id,
+            destination=sai_ipprefix('4444::1/127'),
+            vr_id=self.default_vrf)
+        status = sai_thrift_create_route_entry(
+            self.client, self.route4, next_hop_id=self.nhop)
+        self.assertEqual(status, SAI_STATUS_SUCCESS)
+
+        # test qos map creation
+        dscp_to_tc_1 = sai_thrift_qos_map_t(
+            key=sai_thrift_qos_map_params_t(dscp=2),
+            value=sai_thrift_qos_map_params_t(tc=16))
+        dscp_to_tc_2 = sai_thrift_qos_map_t(
+            key=sai_thrift_qos_map_params_t(dscp=4),
+            value=sai_thrift_qos_map_params_t(tc=24))
+        self.qos_map_list = sai_thrift_qos_map_list_t(
+            count=2, maplist=[dscp_to_tc_1, dscp_to_tc_2])
+        self.qos_map = sai_thrift_create_qos_map(
+            self.client,
+            SAI_QOS_MAP_TYPE_DSCP_TO_TC,
+            map_to_value_list=self.qos_map_list)
+
+    def runTest(self):
+        attr = sai_thrift_get_fdb_entry_attribute(
+            self.client, self.fdb_entry, bridge_port_id=True)
+        self.assertEqual(attr['bridge_port_id'], self.port0_bp)
+
+        for route in [self.route0, self.route1, self.route2,
+                      self.route3, self.route4]:
+            attr = sai_thrift_get_route_entry_attribute(
+                self.client, route, next_hop_id=True)
+            self.assertEqual(attr['next_hop_id'], self.nhop)
+
+        attr = sai_thrift_get_next_hop_attribute(
+            self.client, self.nhop, ip='0.0.0.0')
+        self.assertEqual(attr['ip'].addr.ip4, '10.10.10.1')
+        attr = sai_thrift_get_next_hop_attribute(
+            self.client, self.nhop1, ip='0.0.0.0')
+        self.assertEqual(attr['ip'].addr.ip6, '4444::1')
+
+        attr = sai_thrift_get_neighbor_entry_attribute(
+            self.client, self.neigh_entry, dst_mac_address=True)
+        self.assertEqual(attr['dst_mac_address'], '00:11:22:33:44:66')
+
+        attr = sai_thrift_get_qos_map_attribute(
+            self.client,
+            self.qos_map,
+            map_to_value_list=sai_thrift_qos_map_list_t(
+                count=2, maplist=[]))
+        self.assertEqual(attr['map_to_value_list'].count,
+                            self.qos_map_list.count)
+        for i in range(0, self.qos_map_list.count):
+            self.assertEqual(attr['map_to_value_list'].maplist[i].key.dscp,
+                                self.qos_map_list.maplist[i].key.dscp)
+            self.assertEqual(attr['map_to_value_list'].maplist[i].value.tc,
+                                self.qos_map_list.maplist[i].value.tc)
+
+    def tearDown(self):
+        sai_thrift_remove_qos_map(self.client, self.qos_map)
```

Comment:
Can you update this teardown method as well

Reply:
sure

Reply:
done